src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -893,12 +894,14 @@ private async Task TimerLoop(TimerAwaitable timer)
             ResetTimeout();
             ResetSendPing();
 
+            var inherentKeepAlive = _connectionState.Connection.Features.Get<IConnectionInherentKeepAliveFeature>()?.HasInherentKeepAlive ?? false;
+
             using (timer)
             {
                 // await returns True until `timer.Stop()` is called in the `finally` block of `ReceiveLoop`
                 while (await timer)
                 {
-                    if (DateTime.UtcNow.Ticks > Volatile.Read(ref _nextActivationServerTimeout))
+                    if (!inherentKeepAlive && DateTime.UtcNow.Ticks > Volatile.Read(ref _nextActivationServerTimeout))
```

Comment:
Should the server not ping connections with inherent keep alive? or is this more trouble than it's worth?

Reply:
We already don't send the pings
https://github.com/aspnet/SignalR/blob/810c7cf9ad70d2a12708826100127a085b556a7f/src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs#L415-L419